modules/mobile/app/services/mobile/v2/appointments/presentation_filter.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+# these filters were derived from the web app front end code base
+module Mobile
+  module V2
+    module Appointments
+      class PresentationFilter
+        def initialize(include_pending:)
+          @include_pending = include_pending
+          @now = DateTime.now.utc
+        end
+
+        def user_facing?(appointment)
+          return true if valid_appointment?(appointment) &&
+                         (presentable_upcoming_appointment?(appointment) ||
+                          presentable_past_appointment?(appointment) ||
+                          presentable_cancelled_appointment?(appointment))
+
+          return false unless @include_pending
+
+          presentable_requested_appointment?(appointment)
+        end
+
+        private
+
+        def presentable_upcoming_appointment?(appointment)
+          appointment[:start] >= @now
+        end
+
+        def presentable_past_appointment?(appointment)
+          appointment[:start] < @now && appointment[:status] != 'cancelled'
+        end
+
+        def presentable_cancelled_appointment?(appointment)
+          appointment[:status] == 'cancelled' && appointment[:start] >= 30.days.ago.beginning_of_day
```

Comment:
Some of these conditions look to be good common sense safe guards but others such as only showing cancelled appointments within the last 30 days look to be more of a web UX decision rather then a valid data safe guard. should we run it by product to make sure we want to mirror this?